src/xGadgetron/cGadgetron/gadgetron_data_containers.cpp

```diff
@@ -1140,6 +1143,98 @@ GadgetronImagesVector::set_real_data(const float* data)
+            throw std::runtime_error("GadgetronImagesVector::set_up_geom_info(): read_dir, phase_dir and slice_dir should all be unit vectors.");
+
+        // Calculate the spacing!
+        ISMRMRD::ImageHeader &ih2 = image_wrap(1).head();
+        float projection_of_position_in_slice_dir_1 = ih1.position[0] * ih1.slice_dir[0] +
+                ih1.position[1] * ih1.slice_dir[1] +
+                ih1.position[2] * ih1.slice_dir[2];
+        float projection_of_position_in_slice_dir_2 = ih2.position[0] * ih2.slice_dir[0] +
+                ih2.position[1] * ih2.slice_dir[1] +
+                ih2.position[2] * ih2.slice_dir[2];
+        spacing[2] = std::abs(projection_of_position_in_slice_dir_1 - projection_of_position_in_slice_dir_2);
+
+        // Now for some more checks...
+        // Loop over all images, and
+        // 1. Check that the slice_dir is always constant
+        // 2. Check that spacing is more-or-less constant
+        for (unsigned im=0; im<number()-1; ++im) {
+
+            ISMRMRD::ImageHeader &ih1 = image_wrap( im ).head();
+            ISMRMRD::ImageHeader &ih2 = image_wrap(im+1).head();
+
+            // 1. Check that the slice_dir is always constant
+            for (int dim=0; dim<3; ++dim)
+                if (std::abs(slice_dir[dim]-ih1.slice_dir[dim]) > 1.e-7F)
+                    throw std::runtime_error("GadgetronImagesVector::set_up_geom_info(): Slice direction alters between different slices. Expected it to be constant.");
+
+            // 2. Check that spacing is constant
+            float projection_of_position_in_slice_dir_1 = ih1.position[0] * ih1.slice_dir[0] +
+                    ih1.position[1] * ih1.slice_dir[1] +
+                    ih1.position[2] * ih1.slice_dir[2];
+            float projection_of_position_in_slice_dir_2 = ih2.position[0] * ih2.slice_dir[0] +
+                    ih2.position[1] * ih2.slice_dir[1] +
+                    ih2.position[2] * ih2.slice_dir[2];
+            float new_spacing = std::abs(projection_of_position_in_slice_dir_1 - projection_of_position_in_slice_dir_2);
+            if (std::abs(spacing[2]-new_spacing) > 1.e-5F)
+                throw std::runtime_error("GadgetronImagesVector::set_up_geom_info(): Slice distances alters between slices. Expected it to be constant.");
+        }
+    }
+
+    // Offset
+    VoxelisedGeometricalInfo3D::Offset offset;
+    for (int i=0; i<3; ++i)
+        offset[i] = ih1.position[i];
+
+    // Direction
+    VoxelisedGeometricalInfo3D::DirectionMatrix direction;
+    for (int axis=0; axis<3; ++axis) {
+        direction[0][axis] = ih1.read_dir[axis];
+        direction[1][axis] = ih1.phase_dir[axis];
+        direction[2][axis] = ih1.slice_dir[axis];
+    }
+
+    // Initialise the geom info shared pointer
+    _geom_info_sptr = std::make_shared<VoxelisedGeometricalInfo3D>(
+                VoxelisedGeometricalInfo3D(offset,spacing,size,direction));
```

Comment:
```suggestion
                offset,spacing,size,direction);
```
I believe. I think now we're making a copy and then throwing it away